sdks/csharp/adtIotHub_delete.cs

```diff
@@ -38,21 +49,22 @@ public static class DeleteDeviceInTwinFunc
                     if (opType == "deleteDeviceIdentity")
                     {
                         string deviceId = eventData.Properties["deviceId"] as string;
-                        
-                        // Create Digital Twin client
-                        var cred = new ManagedIdentityCredential(adtAppId);
-                        var client = new DigitalTwinsClient(new Uri(adtInstanceUrl), cred, new DigitalTwinsClientOptions { Transport = new HttpClientTransport(httpClient) });
 
-                        // Find twin based on the original Registration ID
-                        string regID = deviceId; // simple mapping
-                        string dtId = await GetTwinId(client, regID, log);
-                        if (dtId != null)
+                        try
                         {
-                            await DeleteRelationships(client, dtId, log);
+                            // Find twin based on the original Registration ID
+                            BasicDigitalTwin digitalTwin = await client.GetDigitalTwinAsync<BasicDigitalTwin>(deviceId);
```

Comment:
Previous helper function was running a query to find a specific digital twin by its Id, which we can do a lot more efficiently with a direct 'get' call. The downside is that if the twin is not found it'll throw an exception with a 404 status code. So we just catch that and do nothing (as before).